### PR TITLE
Run OhMyREPL keybindings in a fixed world age

### LINKED ISSUE #316: Noticeable lag when importing with ProgressLogging.jl
I'm noticing that there is considerable lag in the REPL when importing OhMyREPL.jl and ProgressLogging.jl on julia 1.9-rc1. The lag isn't in import, but the REPL hangs quite a bit after the second package is loaded (it doesn't matter which order). 

Using:
```
ProgressLogging v0.1.4
OhMyREPL v0.5.20
```

### PR DESCRIPTION
**Pull request by c42f**
Hopefully this should avoid most invalidations caused by loading other packages as discussed in https://github.com/JuliaLang/JuliaSyntax.jl/issues/252.

Also refactor/replace the NEW_KEYBINDINGS global as it's hard to follow where this is added to, and surface a bunch of other uses of global variables like `Base.active_repl` up to the top level init.

I'm not sure whether there's some additional wrapping which is needed around the pirating in `refresh_lines.jl`, etc?

Seems to fix #316 ?

@jakobnissen You've used OhMyREPL a lot more than me, perhaps you could give this a test drive? I'm not entirely sure of all the ways I should be trying to test this.

### CODE REVIEW

**Comment by jakobnissen** [comment]
This PR indeed removes all (noticable) latency even when running with JuliaSyntax 0.3.4 (before Kristoffer's recent type-stability improvements to JuliaSyntax) and importing ProgressLogging.

When using the latest JuliaSyntax, loading ProgressLogging still leads to 277 invalidations before your PR, and 253 with your PR. In either case, latency is not noticable even when the invalidated code is in the REPL - maybe I just can't figure out how to trigger the invalidated code to recompile.

Still, from a latency viewport this is a minor win. If there a no/few downsides, it looks good. If there are any downsides to running in a fixed world age, perhaps a better solution might be to make REPL.jl more type stable.

**Comment by c42f** [comment]
> loading ProgressLogging still leads to 277 invalidations before your PR, and 253 with your PR

Right. This doesn't prevent invalidations in the latest world. But running the key bindings in an older world avoids the compiler having to do work to deal with them. So latency should be much improved.

**Comment by KristofferC** [comment]
Good to go?

**Comment by c42f** [comment]
I've updated this to also fix the world age of the pirated `LineEdit.refresh_line`, and to precompile the implementations of all pirated methods. This should reduce latency a little.

**Comment by c42f** [comment]
Oh I didn't see your comment. Yes, I think it's good to go now :-)

**Comment by c42f** [comment]
Ugh this has gotten stale/conflicted.

I've fixed the conflicts and pushed again.

It would be nice to merge this? it contains improvements to precompilation, along with the `invoke_in_world` stuff.

**Comment by c42f** [comment]
Windows test failures are probably due to https://github.com/JuliaDebug/TerminalRegressionTests.jl/pull/14 ?

**Comment by Moelf** [comment]
@c42f sorry, looks like that PR didn't fix the CI for windows and on nightly, I reverted the merge

let's try this again when the TerminalRegressionTests is fixed?

**Comment by c42f** [comment]
Oh goshdarn this PR is so old now :-/ I will open a new one to re-propose this so it doesn't get lost. I guess.
